src/nixpacks/environment.rs

```diff
@@ -16,19 +17,26 @@ impl Environment {
     pub fn from_envs(envs: Vec<&str>) -> Result<Environment> {
         let mut environment = Environment::default();
         for env in envs {
-            let v: Vec<&str> = env.split('=').collect();
-            if v.len() == 1 {
+            let matches = Regex::new(r"([A-Za-z0-9_-]*)(?:=?)([A-Za-z0-9_-]*)([\s\S]*)")
```

Comment:
```suggestion
            let matches = Regex::new(r"([A-Za-z0-9_-]*)(?:=?)([^=]*)([\s\S]*)")
```
the variable body can be anything that isnt an `=`

Reply:
done